src/amo/reducers/recommendations.js

```diff
@@ -0,0 +1,182 @@
+/* @flow */
+import invariant from 'invariant';
+
+import { createInternalAddon } from 'core/reducers/addons';
+import type { ExternalAddonType } from 'core/types/addons';
+
+export const ABORT_FETCH_RECOMMENDATIONS: 'ABORT_FETCH_RECOMMENDATIONS'
+  = 'ABORT_FETCH_RECOMMENDATIONS';
+export const FETCH_RECOMMENDATIONS: 'FETCH_RECOMMENDATIONS'
+  = 'FETCH_RECOMMENDATIONS';
+export const LOAD_RECOMMENDATIONS: 'LOAD_RECOMMENDATIONS'
+  = 'LOAD_RECOMMENDATIONS';
+
+export type Recommendations = {|
+  addons: Array<ExternalAddonType> | null,
+  fallbackReason: string | null,
+  loading: boolean,
+  outcome: string | null,
+|};
+
+export type RecommendationsState = {|
+  byGuid: {
+    [guid: string]: Recommendations,
+  },
+|};
+
+export const initialState: RecommendationsState = {
+  byGuid: {},
+};
+
+export type AbortFetchRecommendationsParams = {|
+  guid: string,
+|};
+
+type AbortFetchRecommendationsAction = {|
+  type: typeof ABORT_FETCH_RECOMMENDATIONS,
+  payload: AbortFetchRecommendationsParams,
+|};
+
+export const abortFetchRecommendations = ({
+  guid,
+}: AbortFetchRecommendationsParams = {}): AbortFetchRecommendationsAction => {
+  invariant(guid, 'guid is required');
+  return {
+    type: ABORT_FETCH_RECOMMENDATIONS,
+    payload: { guid },
+  };
+};
+
+type FetchRecommendationsParams = {|
+  errorHandlerId: string,
+  guid: string,
+  recommended: boolean,
+|};
+
+export type FetchRecommendationsAction = {|
+  type: typeof FETCH_RECOMMENDATIONS,
+  payload: FetchRecommendationsParams,
+|};
+
+export const fetchRecommendations = ({
+  errorHandlerId,
+  guid,
+  recommended,
+}: FetchRecommendationsParams = {}): FetchRecommendationsAction => {
```

Comment:
Here too, I would not add the default empty object. It's not really clear to me why we have some actions with default empty objects and others without... I would be in favor of not doing this because it would raise a warning/error in development.

Reply:
Agreed. That was code borrowed from another reducer and modified, but with required arguments there's no reason to default to `{}`.